compiler/dsl/src/json_export.rs

```diff
@@ -0,0 +1,2093 @@
+        // Check if we've reached the end position
+        if !found_end && byte_index >= end_byte {
+            end_line = line;
+            end_column = column;
+            found_end = true;
+            break;
+        }
+        
+        // Update line and column based on character
+        if ch == '\n' {
+            line += 1;
+            column = 1;
+        } else {
+            column += 1;
+        }
+    }
+    
+    // If we didn't find the end position, use the final line/column
+    if !found_end {
+        end_line = line;
+        end_column = column;
+    }
+    
+    (start_line, start_column, end_line, end_column)
+}
+
+/// Serializable representation of a symbol table entry.
+#[derive(Debug, Clone, Serialize)]
+pub struct SymbolEntry {
+    pub name: String,
+    pub symbol_type: String,
+    pub scope: String,
+    pub declaration_location: Option<SerializableSourceSpan>,
+    pub data_type: Option<String>,
+    pub is_external: bool,
+}
+
+/// Serializable representation of the symbol table.
+#[derive(Debug, Clone, Serialize)]
+pub struct SymbolTable {
+    pub symbols: std::collections::HashMap<String, SymbolEntry>,
+    pub scopes: Vec<String>,
+}
+
+/// Serializable representation of a comment.
+#[derive(Debug, Clone, Serialize)]
+pub struct CommentInfo {
+    pub text: String,
+    pub style: String, // "block" for (* *) or "line" for //
+    pub location: Option<SerializableSourceSpan>,
+    pub attached_to: Option<String>, // ID of the AST node this comment is attached to
+}
+
+/// Serializable representation of formatting information.
+#[derive(Debug, Clone, Serialize)]
+pub struct FormattingInfo {
+    pub comments: Vec<CommentInfo>,
+    pub whitespace_significant: bool,
+}
+
+impl SymbolTable {
+    pub fn new() -> Self {
+        Self {
+            symbols: std::collections::HashMap::new(),
+            scopes: vec!["global".to_string()],
+        }
+    }
+
+    pub fn add_symbol(&mut self, id: String, entry: SymbolEntry) {
+        self.symbols.insert(id, entry);
+    }
+
+    pub fn add_scope(&mut self, scope: String) {
+        if !self.scopes.contains(&scope) {
+            self.scopes.push(scope);
+        }
+    }
+}
+
+/// Extract unique source file IDs from a Library AST.
+/// This function provides a simple implementation that returns a default file list.
+/// In a full implementation, this would traverse the AST to collect all unique file IDs.
+fn extract_source_files(_library: &Library) -> Vec<String> {
+    // For now, return a default file list
+    // In a complete implementation, this would traverse the AST using the fold pattern
+    // to collect all unique FileId values from SourceSpan fields
+    vec!["<unknown>".to_string()]
+}
+
+/// Extract symbol table information from a Library AST.
+/// This creates a basic symbol table by traversing the AST and collecting
+/// function, function block, and variable declarations.
+fn extract_symbol_table(library: &Library) -> SymbolTable {
```

Comment:
I hadn't finished the symbol table for the project, but there is a start in `analyzer`. Let's ensure that there is only one so that generating JSON will use the actual representation that the compiler uses.